tests/test_tokens.py

```diff
@@ -281,10 +281,19 @@ def test_check_exp(self):
         with self.assertRaises(TokenError):
             token.check_exp('refresh_exp', current_time=current_time + timedelta(days=2))
 
+    def test_check_token_not_expired_if_in_leeway(self):
+        token = MyToken()
+        token.set_exp('refresh_exp', lifetime=timedelta(days=1))
+
+        datetime_in_leeway = token.current_time + timedelta(days=1)
+
+        with self.assertRaises(TokenError):
+            token.check_exp('refresh_exp', current_time=datetime_in_leeway)
+
         # a token 1 day expired is valid if leeway is 2 days
-        token._token_backend.leeway = timedelta(days=2).total_seconds()
-        token.check_exp('refresh_exp', current_time=current_time + timedelta(days=1))
-        token._token_backend.leeway = 0
+        token.get_token_backend().leeway = timedelta(days=2).total_seconds()
```

Comment:
What do you think of using the `@property` decorator in the `get_token_backend` method?

ex:)

 tokens.py
```python 

@property
def token_backend(self): 
    # Backward compatibility.
    return self.get_token_backend()
 
```

test_tokens.py
```python
token.token_backend.leeway = timedelta(days=2).total_seconds()
token.check_exp("refresh_exp", current_time=datetime_in_leeway)
token.token_backend.leeway = 0
```

Reply:
@Andrew-Chen-Wang @totycro

Reply:
Looks good to me, I've introduced the property in the branch and for now only use it in the test. It could then be used throughout the code.

I've moved the implementation of `get_token_backend` to the property and use the property in `get_token_backend` to signify that the property should now be the default way to go, and `get_token_backend` is only compatibility.